src/pools/pool.ts

```diff
@@ -1,10 +1,17 @@
+import type { IWorker } from './abstract-pool'
+import { AbstractPool } from './abstract-pool'
+
 /**
  * Contract definition for a poolifier pool.
  *
  * @template Data Type of data sent to the worker. This can only be serializable data.
  * @template Response Type of response of execution. This can only be serializable data.
  */
-export interface IPool<Data = unknown, Response = unknown> {
+export interface IPool<
+  Worker extends IWorker,
```

Comment:
`IPool` doesn't make use of `Worker` so we don't need it here

Reply:
It's needed to allow to properly reuse the same typed implementation for all callbacks on worker-threads or cluster pool, dynamic of fixed.